src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -156,14 +163,28 @@ def train_job(train_cfg, train_dmatrix, val_dmatrix, model_dir, is_master):
     if val_dmatrix is not None:
         watchlist.append((val_dmatrix, 'validation'))
 
+    xgb_model, iteration = train_utils.load_checkpoint(checkpoint_dir)
+    if not is_master:
```

Comment:
Why is this necessary? Alternatively, when does the model get synced between hosts for the resumed job?

Reply:
When Spot resumes a host, it downloads the checkpoints to `/opt/ml/checkpoints` **before** it starts a fresh training job. Line 166 checks the checkpoint directory to see if there are any checkpoint files. If there are, it means the training was interrupted previously so we load from the latest checkpoint. If not, we start from iteration 0.